sdks/java/io/amazon-web-services/src/test/java/org/apache/beam/sdk/io/aws/sns/SnsIOTest.java

```diff
@@ -114,7 +118,10 @@ public void testDataWritesToSNS() {
 
   @Test
   public void testRetries() throws Throwable {
+    thrown.expect(IOException.class);
```

Comment:
Is it an exact exception that we expect?

Reply:
No, it's just important to not accept any Exception! Otherwise the AssertionError thrown by `snsWriterFnLogs.verifyWarn` fullfills this, which is kind of pointless. But that's exactly what happened, because it was checking the wrong logger